packages/util/src/util.ts

```diff
@@ -197,3 +200,109 @@ export const flattenObj = (obj: object, prefix = ''): Record<string, unknown> =>
     }
     return flattenedObj
 }
+
+// https://stackoverflow.com/questions/63116039/camelcase-to-kebab-case
+export const kebabCase = (str: string) =>
+    str.replace(/[A-Z]+(?![a-z])|[A-Z]/g, ($, ofs) => (ofs ? '-' : '') + $.toLowerCase())
+
+export type MergeOptions = {
+    atomicArrays?: boolean
+}
+
+// Merge two objects or arrays together.
+// Nesting can be infinitely deep.
+// Arrays can be homogeneous or hetrogeneous.
+// The destination object/array is mutated directly.
+// Arrays can be merged in two ways:
+// - update (default): replace elements as required and extend array as required, e.g. [1,2,3] + [4,5] = [4,5,3]
+// - replace: treat the array as a primitive value and simply replace as-is, e.g. [1,2,3] + [4,5] = [4,5]
+// The 'atomicArrays' option controls whether arrays are treated as primitives or not. E.g. atomicArrays=true is the 'replace' strategy, atomicArrays=false is the 'update' strategy.
+// This method treats arrays as an object with numeric keys and merged using the object merge strategy.
+export function merge<T extends object | A[], U extends object | B[], A, B>(
+    dest: T,
+    src: U,
+    options?: MergeOptions
+): T & U {
+    const atomicArrays = options?.atomicArrays
+    // maintain a queue of object sources/destinations to merge
+    const queue: {
+        src: unknown
```

Comment:
Can we infer types of src and dest or is it better to leave as unknown?

Reply:
it's better to leave these as unknown as we don't know what type we're handling. Usually it'll be an array or an object, but it may be a primitive if the user passes in parameters for that. I did have types in here but the bounds became super complex, so I switched to unknown because this fn should handle anything and everything